### PR TITLE
Use docker for GPU builds to test against multiple CUDA versions

### PR DESCRIPTION
**Pull request by beauby**
Adding CI jobs with CUDA 9.2, 10.1, 10.2, 11.0, and 11.1.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -62,46 +62,39 @@ jobs:
...
-          command: |
-            cd build
-            sudo make install
+              docker build -t tensorpipe -f .circleci/Dockerfile.cuda<< parameters.cuda_version >> .
+              docker run --gpus all tensorpipe sh -c "
+                mkdir build && cd build &&
+                cmake ../ \
+                  -DCMAKE_C_FLAGS=\"-Werror -Wno-deprecated-declarations\" \
+                  -DCMAKE_CXX_FLAGS=\"-Werror -Wno-deprecated-declarations\" \
+                  -DCUDA_NVCC_FLAGS=\"-gencode arch=compute_61,code=sm_61\" \
```
Comment (lw) [inline .circleci/config.yml:88]: This line is new, why is it needed?

**Review by lw — APPROVED** [review]
Very nice! Thanks!

Could we also add CUDA 9.2 and 11.0? They are two of the versions supported by PyTorch (as of v1.7, not sure if this will change in v1.8) so I think we should cover them too.

File: .circleci/config.yml (hunk 1 above)
Comment (beauby) [reply .circleci/config.yml:88]: It is not needed strictly speaking, but it allows for generating device code optimized for the GPU type available in CircleCI instances.

**Comment by lw** [comment]
The clang-format issue has been fixed in master, the TSAN one seems unrelated and I've opened https://github.com/pytorch/tensorpipe/issues/262 to track it.

**Pull request closed without merging**